### PR TITLE
fastcgi: Add timeouts support to Caddyfile adapter

### LINKED ISSUE #3841: reverseproxy: Add read_timeout/send_timeout options for HTTP transport
Per the doc of [reverse_proxy](https://caddyserver.com/docs/caddyfile/directives/reverse_proxy) directive, the timeout options available for transports are:
- HTTP: dial_timeout, tls_timeout
- FastCGI: dial_timeout, read_timeout, write_timeout (not documented, but found [in code](https://github.com/caddyserver/caddy/blob/937ec342010894f7a6239883b10f6a107ff82c9f/modules/caddyhttp/reverseproxy/fastcgi/fastcgi.go#L68-L75))

For the same reason listed [here](https://github.com/caddyserver/caddy/issues/2113#issuecomment-408840525), the read_timeout (like Nginx's [proxy_read_timeout](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_read_timeout)) and send_timeout (like Nginx's  [proxy_send_timeout](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_send_timeout)) options are also very useful for HTTP transport.

### PR DESCRIPTION
**Pull request by francislavoie**
Discussed in #3841

Nothing fancy here, just wiring up some timeouts.

### CODE REVIEW

File: caddytest/integration/caddyfile_adapt/php_fastcgi_index_off.txt [hunk 1]
```diff
@@ -7,6 +7,9 @@ php_fastcgi localhost:9000 {
     env VAR2 value2
     root /var/www
     index off
+	dial_timeout 3s
+	read_timeout 10s
+	write_timeout 20s
```
Comment (mholt) [inline caddytest/integration/caddyfile_adapt/php_fastcgi_index_off.txt:12]: For some reason, all other indents are spaces, not tabs, so these look weird. Can we normalize this file to use tabs?

**Review by mholt — CHANGES_REQUESTED** [review]
Just a nit-pick I suppose; also, it's really a bummer that the parsing logic has to be duplicated here (I get why, but... sad face).

File: caddytest/integration/caddyfile_adapt/php_fastcgi_index_off.txt (hunk 1 above)
Comment (francislavoie) [reply caddytest/integration/caddyfile_adapt/php_fastcgi_index_off.txt:12]: Oh, oops, ok

**Review by mholt — APPROVED** [review]

**Comment by mholt** [comment]
Thank you!! Appreciate your help with this.

**Pull request merged**